drivers/flash/flash_stm32h7x.c

```diff
@@ -102,6 +101,65 @@ static __unused int write_optb(const struct device *dev, uint32_t mask,
 	return 0;
 }
 
+static __unused int write_wp_reg(const struct device *dev, uint32_t mask, uint32_t value,
+				 uint32_t bank)
+{
+	FLASH_TypeDef *regs = FLASH_STM32_REGS(dev);
+	int rc;
+#ifdef DUAL_BANK
+	uint32_t number_of_banks = 2;
+#else  /* DUAL_BANK */
+	uint32_t number_of_banks = 1;
+#endif /* DUAL_BANK */
+#ifdef CONFIG_SOC_SERIES_STM32H7RSX
+	volatile uint32_t *cur =
+		(volatile uint32_t *)((uint32_t)regs + offsetof(FLASH_TypeDef, WRPSR));
```

Comment:
We should use `uintptr_t` (or `intptr_t`) if you want to keep pointer as a variable.

Reply:
This definitely makes sense :-)